lib/mayaUsd/undo/OpUndoItems.h

```diff
@@ -403,6 +404,46 @@ class SelectionUndoItem : public OpUndoItem
     MGlobal::ListAdjustment _selMode;
 };
 
+//------------------------------------------------------------------------------
+// UfeSelectionUndoItem
+//------------------------------------------------------------------------------
+
+/// \class UfeSelectionUndoItem
+/// \brief Record data needed to undo or redo select nodes sub-operations.
+class UfeSelectionUndoItem : public OpUndoItem
```

Comment:
Added a new UfeSelectionUndoItem class.  Arguably the existing SelectionUndoItem should be transformed to hold a Ufe::Selection, because MSelectionList cannot hold Ufe::SceneItem's.  At that point we can simply remove UfeSelectionUndoItem.  This work is out of scope for this pull request.